### ISSUE TITLE
Nethack 3.6.2 on macOS - unable to [un]compress bone files

### ISSUE THREAD

**Issue by j3k0** [NetHack/NetHack#191]
Since I updated to the latest version (3.6.2) on macOS, I keep getting the following error:

```
compress: /dev/stdin: Permission denied
Unable to uncompress bonD0.4
```

<img width="752" alt="Screen Shot 2019-05-18 at 06 21 25" src="https://user-images.githubusercontent.com/191881/57964157-0c6bd100-7939-11e9-93e3-97197a33a3f2.png">

Happens when I go downstairs.

It never happened with Nethack 3.6.1

**Comment by bhaak** [comment]
I can't reproduce this on 10.12.6.

What's your macOS version and how do the bones files permission look like:
`ls -l /Library/NetHack/nethackdir/bon*`

**Comment by j3k0** [comment]
* macOS 10.14.1

```
$ ls -la /Library/NetHack/nethackdir/
total 2976
drwxrwxr-x  19 root  games      608 May 18 09:55 ./
drwxr-xr-x   6 root  wheel      192 May 13 18:22 ../
-rw-rw----   1 jeko  games    26924 May 18 09:55 alock.0
-rw-rw----   1 jeko  games    20969 May 18 09:55 alock.1
-rw-rw----   1 root  games     5008 Nov 23 13:07 bonD0.10.Z
-rw-rw----   1 root  games     7660 Nov 30 23:41 bonD0.13.Z
-rw-rw----   1 root  games     5038 Feb 22 20:44 bonD0.4.Z
-rw-rw----   1 root  games     4189 May  7 10:09 bonD0.5.Z
-rw-rw----   1 root  games     5374 May  9 10:51 bonD0.6.Z
-rw-r--r--   1 root  games     4875 Dec 22 00:06 license
-rw-rw-r--   1 root  games       89 May 18 06:53 logfile
-rw-r--r--   1 root  games  1380333 May  9 08:58 nhdat
-rw-rw-r--   1 root  games        0 May  9 08:58 perm
-rw-rw-r--   1 root  games       89 May 18 06:53 record
-rwxr-xr-x   1 root  games     1303 May  9 08:58 recover.pl
drwxrwxr-x   2 root  games       64 May 13 18:22 save/
-rw-r--r--   1 root  games    14744 Dec 22 00:06 symbols
-rw-rw-r--   1 root  games     5527 May  9 08:58 sysconf
-rw-rw-r--   1 root  games      329 May 18 06:53 xlogfile
```

I can assure you I would never run nethack as root. If I recall correctly, previous version was also installed through the binaries linked on nethack's website. Maybe the files were already put in place by the installer?

My user `jeko` isn't a member of the `games` group (none of the users on my mac are, it seems to be the default behavior).

---

edit, if helpful for a fix: groups that don't start with a `_` that all users are a member of: `staff everyone localaccounts`

**Comment by nhmall** [comment]
Was this a downloaded binary or did you build your own, and if the latter which sys/unix/hints file did you use?  I'm only asking because I noticed this past commit 
[https://github.com/NetHack/NetHack/commit/f38dc942c0b44046ca5745127fbb3ef3c0074623#diff-31e519f726e95e622b236ced649420be](url)

**Comment by bhaak** [comment]
I can now reproduce that. Having bones files from a previous installation, during the installation of the new version the user of the bones files will be changed to `root`. And as the bones files are only user and group read/writeable, NetHack can see but not read or write the bones files as the permission to so is denied (note: similar problem as was fixed for saves in f38dc942c0b44046ca5745127fbb).

For you the workaround is to do `sudo /bin/chmod 666 /Library/NetHack/nethackdir/bon*`

This will prompt for your password as it needs elevated user rights. New bones files are not affected.

**Comment by j3k0** [comment]
Thanks @bhaak for the follow-up, after changing permissions I do not seem to see warnings anymore (but I didn't play for long).
